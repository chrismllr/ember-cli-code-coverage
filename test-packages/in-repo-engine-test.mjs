import { execa } from 'execa';
import { it, describe } from 'vitest';
import setupTestDir, { assertCoverageExists } from './utils.mjs';

const APP_DIR = 'my-app-with-in-repo-engine';

describe('in-repo engine coverage generation', function () {
  it('runs coverage on in-repo engine', async function () {
    let env = { COVERAGE: 'true' };
    let buildPath = await setupTestDir(APP_DIR, env, {});
    await execa('pnpm', ['ember' , 'test','--test-port=0'], { cwd: buildPath, env });
    await assertCoverageExists(`${buildPath}/coverage`);
  });
});
