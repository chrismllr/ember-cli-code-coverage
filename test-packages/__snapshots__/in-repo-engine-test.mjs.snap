// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`in-repo engine coverage generation > runs coverage on in-repo engine 1`] = `
{
  "app/app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
    "statements": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/router.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 25,
      "skipped": 0,
      "total": 4,
    },
    "statements": {
      "covered": 1,
      "pct": 25,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/utils/my-covered-util-app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "statements": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
  },
  "app/utils/my-uncovered-util-app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "statements": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
  },
  "lib/my-in-repo-engine/addon/engine.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 3,
      "pct": 100,
      "skipped": 0,
      "total": 3,
    },
    "statements": {
      "covered": 3,
      "pct": 100,
      "skipped": 0,
      "total": 3,
    },
  },
  "lib/my-in-repo-engine/addon/resolver.js": {
    "branches": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
    "statements": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
  },
  "lib/my-in-repo-engine/addon/routes.js": {
    "branches": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
    "statements": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 0,
    },
  },
  "lib/my-in-repo-engine/addon/utils/my-covered-util.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "statements": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
  },
  "lib/my-in-repo-engine/addon/utils/my-uncovered-util.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "statements": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
  },
  "total": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "branchesTrue": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 2,
      "pct": 33.33,
      "skipped": 0,
      "total": 6,
    },
    "lines": {
      "covered": 10,
      "pct": 66.66,
      "skipped": 0,
      "total": 15,
    },
    "statements": {
      "covered": 10,
      "pct": 66.66,
      "skipped": 0,
      "total": 15,
    },
  },
}
`;
